pkg/apiserver/resthandler.go

```diff
@@ -446,6 +417,93 @@ func PatchResource(r rest.Patcher, scope RequestScope, typer runtime.ObjectTyper
 
 		write(http.StatusOK, scope.APIVersion, scope.Codec, result, w, req.Request)
 	}
+
+}
+
+// patchResource divides PatchResource for easier unit testing
+func patchResource(ctx api.Context, timeout time.Duration, versionedObj runtime.Object, r rest.Patcher, name string, patchType api.PatchType, patchJS []byte, namer ScopeNamer, codec runtime.Codec) (runtime.Object, error) {
```

Comment:
sorry-- I meant rename 'r' to 'patcher'.

On Thu, Oct 1, 2015 at 5:03 PM, David Eads <notifications@github.com> wrote:

> In pkg/apiserver/resthandler.go
> <https://github.com/kubernetes/kubernetes/pull/14758#discussion_r40981191>
> :
>
> > @@ -446,6 +417,93 @@ func PatchResource(r rest.Patcher, scope RequestScope, typer runtime.ObjectTyper
> >
> >  		write(http.StatusOK, scope.APIVersion, scope.Codec, result, w, req.Request)
> >  	}
> > +
> > +}
> > +
> > +// patchResource divides PatchResource for easier unit testing
> > +func patchResource(ctx api.Context, timeout time.Duration, versionedObj runtime.Object, r rest.Patcher, name string, patchType api.PatchType, patchJS []byte, namer ScopeNamer, codec runtime.Codec) (runtime.Object, error) {
>
> this function is too long for 'r'. 'patcher'?
>
> The different arguments (most notably the request and response) make this
> easier to unit test.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/14758/files#r40981191>.
>